web-app/react-app/actions/identity.js

```diff
@@ -0,0 +1,89 @@
+// @flow
+
+import createLink from '../helpers/createLink'
+import {error, success} from 'react-notification-system-redux'
+import axios from 'axios/index'
+import {parseError} from './utils/parseApiResponse'
+import type {ErrorInUi} from '../flowtype/common-types'
+import ownWeb3 from '../components/ProfilePage/IdentityHandler/web3'
+import type {IntegrationKey} from '../flowtype/integration-key-types'
+
+const apiUrl = 'api/v1/integration_keys'
+
+export const CREATE_IDENTITY_REQUEST = 'CREATE_IDENTITY_REQUEST'
+export const CREATE_IDENTITY_SUCCESS = 'CREATE_IDENTITY_SUCCESS'
+export const CREATE_IDENTITY_FAILURE = 'CREATE_IDENTITY_FAILURE'
+
+export const createIdentity = (integrationKey: IntegrationKey) => (dispatch: Function) => {
+    dispatch(createIdentityRequest(integrationKey))
+    if (!ownWeb3) {
+        dispatch(createIdentityFailure({
+            message: 'MetaMask browser extension is not installed'
+        }))
+        dispatch(error({
+            title: 'Create identity failed',
+            message: 'MetaMask browser extension is not installed',
+            autoDismiss: 0,
+        }))
+        return
+    }
+    if (ownWeb3.eth.defaultAccount == null) {
+        dispatch(createIdentityFailure({
+            message: 'MetaMask browser extension is locked'
+        }))
+        dispatch(error({
+            title: 'Create identity failed',
+            message: 'MetaMask browser extension is locked',
+            autoDismiss: 0,
+        }))
+        return
+    }
+    let challenge
+    axios.post(createLink('api/v1/login/challenge')).then(({data}) => {
+        challenge = data
+        return ownWeb3.eth.personal.sign(data.challenge, ownWeb3.eth.defaultAccount)
+    })
+        .then((signature) => {
+            const input = {
+                ...integrationKey,
+                signature: signature,
+                address: ownWeb3.eth.defaultAccount,
+                challenge: {
+                    ...challenge
+                },
+            }
+            return axios.post(createLink(apiUrl), input)
```

Comment:
Could we use `createIntegrationKey` from `actions/integrationKey.js` here? (Now the identities are removed with `deleteIntegrationKey` action so it would follow the line). it would then go somehow like this:
```
return dispatch(createIntegrationKey(input))
    .then(...)
```
The only problem is the error handler (which now triggers a notification so then there would be two notifications). Maybe I'll think about this a while and then we could have a call about it?

Reply:
If we try to reuse the code from createIntegrationKey() it leads to more complexity. So I don't think that a little duplication here hurts at all?